_episodes/04-package-structure.md

```diff
@@ -10,8 +10,505 @@ keypoints:
+
+Create a new directory called io within the package and create two new files (using your operating system or `touch` as prefered):
+`pdb.py`
+```
+"""
+Functions for manipulating pdb files.
+"""
+def open_pdb(file_location):
+    """Open and read coordinates and atom symbols from a pdb file.
+
+    The pdb file must specify the atom elements in the last column, and follow
+    the conventions outlined in the PDB format specification.
+    
+    Parameters
+    ----------
+    file_location : str
+        The location of the xyz file to read in.
+    
+    Returns
+    -------
+    coords : np.ndarray
+        The coordinates of the xyz file.
+    symbols : list
+        The atomic symbols of the xyz file.
+
+    """
+
+    with open(file_location) as f:
+        data = f.readlines()
+    
+    coordinates = []
+    symbols = []
+    
+    for line in data:
+        if 'ATOM' in line[0:6] or 'HETATM' in line[0:6]:
+            symbols.append(line[76:79].strip())
+            
+            coords = [float(x) for x in line[30:55].split()]
+            coordinates.append(coords)
+        
+    coords = np.array(coordinates)
+    symbols = np.array(symbols)
+
+    return symbols, coords
+```
+{: .language-python}
+
+and `xyz.py`
+```
+"""
+Functions for manipulating xyz files.
+"""
+def open_xyz(file_location):
+    """Open and read coordinates and atom symbols from a xyz file.
+
+    Parameters
+    ----------
+    file_location : str
+        The location of the xyz file to read in.
+    
+    Returns
+    -------
+    coords : np.ndarray
+        The coordinates of the xyz file.
+    symbols : np.ndarray
+        The atomic symbols of the xyz file.
+
+    """
+    
+    # Open an xyz file and return symbols and coordinates.
+    xyz_file = np.genfromtxt(fname=file_location, skip_header=2, dtype='unicode')
+    symbols = xyz_file[:,0]
+    coords = (xyz_file[:,1:])
+    coords = coords.astype(np.float)
+    return symbols, coords
+
+def write_xyz(file_location, symbols, coordinates):
+    """Write an xyz file.
+
+    Parameters
+    ----------
+    file_location : str
+        The location of the xyz file to read in.
+    coords : np.ndarray
+        The coordinates to write to the xyz file.
+    symbols : list
+        The atomic symbols to write to the xyz file.
+    """
+    
+    num_atoms = len(symbols)
+    
+    with open(file_location, 'w+') as f:
+        f.write('{}\n'.format(num_atoms))
+        f.write('XYZ file\n')
+        
+        for i in range(num_atoms):
+            f.write('{}\t{}\t{}\t{}\n'.format(symbols[i], 
+                                              coordinates[i,0], coordinates[i,1], coordinates[i,2]))
+```
+{: .language-python}
```

Comment:
One comment here is that I believe that they will be adding docstrings in the next lesson, so none of the functions should have docstrings yet :)

Reply:
Will none of them have docstrings at this point? or will there be 1 or 2 that already have them?

Reply:
Actually none of them will. Unless we'd like to include some at the starting point. I may write in comments (`#`) a basic explanation of each function.